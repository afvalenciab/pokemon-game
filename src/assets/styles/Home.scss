@import url('https://fonts.googleapis.com/css?family=Space+Mono:400,700&display=swap');

body {
  margin: 0;
  font-family: 'Space Mono', monospace;
}

.home--container {
  display: grid;
  grid-template: repeat(2, 1fr) / repeat(4, 1fr);
  grid-column-gap: 5px;
  grid-row-gap: 15px;
  min-height: 100vh;
  padding: 10px;
  background-color: #f8f8fa;
  box-sizing: border-box;
  grid-template-areas:
    'title . player-one  player-one'
    'player-two player-two . options';
}

.player--one {
  grid-area: player-one;
}

.player--two {
  grid-area: player-two;
}