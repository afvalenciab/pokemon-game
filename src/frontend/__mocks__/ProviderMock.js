import React from 'react';
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import initialState from './initialState';
import reducer from '../reducers';

const store = createStore(reducer, initialState, compose(applyMiddleware(thunk)));
const history = createBrowserHistory();

const ProviderMock = (props) => {
  return(
    <Provider store={store}>
      <Router history={history}>
        {props.children}
      </Router>
    </Provider>
  );
};

export default ProviderMock;