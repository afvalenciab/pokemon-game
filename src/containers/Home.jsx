import React from 'react';
import '../assets/styles/Home.scss';
import Header from '../components/Header';
import Player from '../components/Player';
import Game from '../components/Game';
import Footer from '../components/Footer';
import { PlayerContext } from '../context/context';

const Home = () => {
  return (
    <main className='home--container'>
      <Header />
      <section className='player--one'>
        <PlayerContext.Provider value='PlayerOne'>
          <Player/>
        </PlayerContext.Provider>
        {/* <Game /> */}
      </section>
      <section className='player--two'>
        <PlayerContext.Provider value='PlayerTwo'>
          <Player/>
        </PlayerContext.Provider>
        {/* <Game /> */}
      </section>
      <Footer />
    </main>
  );
};

export default Home;
