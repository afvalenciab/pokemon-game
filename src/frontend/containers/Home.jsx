import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { setStatusGame } from '../actions';
import { PlayerContext } from '../context/context';
import Header from '../components/Header';
import Player from '../components/Player';
import Game from '../components/Game';
import Footer from '../components/Footer';
import LoadingPokemon from '../components/LoadingPokemon';
import Modal from '../containers/Modal';
import Winner from '../components/Winner';
import '../assets/styles/Home.scss';

const Home = props => {
  const { lang } = props.match ? props.match.params : '';
  const {
    pokemonPlayerOne,
    pokemonPlayerTwo,
    loadingPokemonOne,
    loadingPokemonTwo,
    setStatusGame,
    statusGame,
    currentPlayer
  } = props;

  const [winner, setWinner] = useState({
    pokemon: {}
  });

  const handleRestartGame = () => {
    setStatusGame('stop');
    setWinner({
      pokemon: {}
    });
  };

  const handlePlayGame = () => {
    setStatusGame('game');
  };

  useEffect(() => {
    if (pokemonPlayerOne) {
      if (pokemonPlayerOne.lifeBar <= 0) {
        setWinner({
          pokemon: pokemonPlayerTwo
        });
      }
    }
  }, [pokemonPlayerOne]);

  useEffect(() => {
    if (pokemonPlayerTwo) {
      if (pokemonPlayerTwo.lifeBar <= 0) {
        setWinner({
          pokemon: pokemonPlayerOne
        });
      }
    }
  }, [pokemonPlayerTwo]);

  return (
    <>
      <main className='home--container'>
        <Header />
        <section className='player--one'>
          <PlayerContext.Provider value='PlayerOne'>
            {!pokemonPlayerOne && !loadingPokemonOne && <Player lang={lang} />}

            {!pokemonPlayerOne && loadingPokemonOne && <LoadingPokemon />}

            {pokemonPlayerOne && !loadingPokemonOne && (
              <Game {...pokemonPlayerOne} lang={lang} />
            )}
          </PlayerContext.Provider>
          <div className={`overlay__one ${currentPlayer === 'PlayerOne' ? '' :'is-disable' }`}></div>
        </section>
        <section className='player--two'>
          <PlayerContext.Provider value='PlayerTwo'>
            {!pokemonPlayerTwo && !loadingPokemonTwo && <Player lang={lang} />}

            {!pokemonPlayerTwo && loadingPokemonTwo && <LoadingPokemon />}

            {pokemonPlayerTwo && !loadingPokemonTwo && (
              <Game {...pokemonPlayerTwo} lang={lang} />
            )}
          </PlayerContext.Provider>
          <div className={`overlay__two ${currentPlayer === 'PlayerTwo' ? '' : 'is-disable'}`}></div>
        </section>
        <Footer hanldeClickRestart={handleRestartGame} handleClickPlay={handlePlayGame} statusGame={statusGame} lang={lang}/>
      </main>
      <div className={`overlay__home ${statusGame === 'stop' ? 'is-active':''}`}></div>
      {Object.keys(winner.pokemon).length > 0 && (
        <Modal>
          <Winner {...winner.pokemon} hanldeClickRestart={handleRestartGame} lang={lang} />
        </Modal>
      )}
    </>
  );
};

const mapStateToProps = state => {
  return {
    pokemonPlayerOne: state.pokemonPlayerOne,
    pokemonPlayerTwo: state.pokemonPlayerTwo,
    loadingPokemonOne: state.loadingPokemonOne,
    loadingPokemonTwo: state.loadingPokemonTwo,
    statusGame: state.statusGame,
    currentPlayer: state.currentPlayer,
  };
};

const mapDispatchToProps = {
  setStatusGame
};

export default connect(mapStateToProps, mapDispatchToProps)(Home);
