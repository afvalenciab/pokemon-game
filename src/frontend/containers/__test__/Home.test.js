import React from 'react';
import { create } from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import Home from '../Home';

describe('Home testing', () => {
  test('Match Snapshot', () => {
    const home = create(
      <ProviderMock>
        <Home />
      </ProviderMock>
    );
    expect(home.toJSON()).toMatchSnapshot();
  });
});