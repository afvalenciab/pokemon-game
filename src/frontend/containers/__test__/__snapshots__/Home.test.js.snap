// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home testing Match Snapshot 1`] = `
Array [
  <main
    className="home--container"
  >
    <header
      className="home--title"
    >
      <figure>
        <img
          alt="Pokemon Logo"
          src="test-file-stubs"
        />
      </figure>
    </header>
    <section
      className="player--one"
    >
      <section
        className="home--player"
      >
        <h2>
          Selecciona un Pokémon
        </h2>
        <div
          className="player--search"
        >
          <form
            onSubmit={[Function]}
          >
            <input
              onChange={[Function]}
              placeholder="Buscar Pokémon por nombre o Id"
              type="text"
            />
            <button
              type="submit"
            >
              ⟶
            </button>
          </form>
        </div>
      </section>
      <div
        className="overlay__one is-disable"
      />
    </section>
    <section
      className="player--two"
    >
      <section
        className="home--player"
      >
        <h2>
          Selecciona un Pokémon
        </h2>
        <div
          className="player--search"
        >
          <form
            onSubmit={[Function]}
          >
            <input
              onChange={[Function]}
              placeholder="Buscar Pokémon por nombre o Id"
              type="text"
            />
            <button
              type="submit"
            >
              ⟶
            </button>
          </form>
        </div>
      </section>
      <div
        className="overlay__two is-disable"
      />
    </section>
    <footer
      className="home--options is-active"
    >
      <div
        className="pokeball"
      >
        <button
          className="button--play"
          onClick={[Function]}
        >
          Jugar
        </button>
        <button
          className="button--restart"
          onClick={[Function]}
        >
          Reiniciar
        </button>
      </div>
      <figure>
        <img
          alt="pokemon snorlax"
          src="test-file-stubs"
        />
      </figure>
    </footer>
  </main>,
  <div
    className="overlay__home is-active"
  />,
]
`;
